src/legacy/core_plugins/vis_type_markdown/public/settings_options.tsx

```diff
@@ -22,7 +22,7 @@ import { EuiPanel } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 
 import { VisOptionsProps } from 'ui/vis/editors/default';
-import { RangeOption, SwitchOption } from '../../vis_type_vislib/public/components';
+import { RangeOption, SwitchOption } from '../../vis_type_vislib/public';
```

Comment:
Just realized these visualization types depend on vislib (at least for static code) - we should put them in a central place so visualizations don't depend on each other (not necessarily in this PR though). I probably missed this in the vislib shimming PR.

I saw `RangeOption`, `SwitchOption`, `ColorModes`, `Labels`, `Styles` and `NumberInputOption`.

Looks like `RangeOption`, `SwitchOption` and `NumberInputOption` are a thin layer on top of eui form components. @streamich Would they fit into `kibana_react`? They are not really chart specific.

`ColorModes`, `Labels` and `Styles` are types - I would say they don't really have to be shared, but _if_ they are shared I think they should go into `visualizations` @ppisljar what do you think?

Reply:
Ya I wasn't sure what to do with these either.

Reply:
ColorModes is a setting specific to metric visualization ... not sure about Labels and Styles, but i am guessing its also settings limited to just a few visualizations ?

Reply:
Let's duplicate the types and make them internal to the plugins they are used in.

For the components it seems like they do not really fit into `kibana_react` - let's make them a part of the new "charts" plugin for now (doesn't have to happen in this PR) - it's a bit of a different beast, but these forms are also useful for all chart types to make the forms consistent, so it's the best home for them now - they are slightly too complex to just duplicate them IMHO.

Reply:
I'll update that before merging

Reply:
Joe I'll update the types and components is another pr